src/Commands/CommandGroup.cs

```diff
@@ -0,0 +1,209 @@
+
+        var sourceChannelResult = await channelApi.GetChannelAsync(sourceChannelId);
+        if (!sourceChannelResult.IsSuccess)
+            return Result.FromError(sourceChannelResult.Error);
+
+        if (sourceChannelResult.Entity.GuildID == null)
+            return Result.FromError(new NotFoundError("This command can only be used inside a server."));
+
+        var guildId = sourceChannelResult.Entity.GuildID.Value;
+
+        var channelsResult = await guildApi.GetGuildChannelsAsync(guildId);
+        if (!channelsResult.IsSuccess)
+            return Result.FromError(channelsResult.Error);
+
+        var destinationChannel = channelsResult.Entity
+            .FirstOrDefault(c => string.Equals(c.Name.Value, destinationChannelName, StringComparison.OrdinalIgnoreCase));
+
+        if (destinationChannel == null)
+            return Result.FromError(new NotFoundError($"Channel '{destinationChannelName}' not found."));
+
+        var destinationChannelId = destinationChannel.ID;
+
+
+        if (!context.TryGetUserID(out var userId))
+            return Result.FromError(new InvalidOperationError("Could not determine the user ID."));
+
+        var botPermissionsResult = await guildApi.GetGuildMemberAsync(guildId, userId);
+        if (!botPermissionsResult.IsSuccess)
+            return Result.FromError(botPermissionsResult.Error);
+
+
+        var botPermissions = botPermissionsResult.Entity.Roles
+            .Select(roleId => guildApi.GetGuildRolesAsync(guildId, CancellationToken.None).Result.Entity.First(role => role.ID == roleId).Permissions);
+
+
+        DiscordPermission permissions = default;
+
+        foreach (var set in botPermissions)
+        {
+            if (set.HasPermission(DiscordPermission.SendMessages))
+            {
+                permissions = DiscordPermission.SendMessages;
+                break;
+            }
+        }
+
+        if (permissions == default)
+            return Result.FromError(new InvalidOperationError("Bot does not have permission to send messages in the destination channel."));
+
+        var sourceChannelMention = $"<#{sourceChannelId}>";
+        var destinationChannelMention = $"<#{destinationChannelId}>";
+
+        var outPortalEmbed = new Embed(
+            Title: "🌌 Portal Created! 🌌",
+            Description: $"Click below to travel to {destinationChannelMention}",
+            Colour: new Optional<Color>(Color.Teal),
+            Thumbnail: new Optional<IEmbedThumbnail>(new EmbedThumbnail(Url: "https://cdn.discordapp.com/attachments/642818541426573344/960339465891631186/b.png"))
+        );
+
+        var sourceMessage = await channelApi.CreateMessageAsync(
+            sourceChannelId,
+            embeds: new[] { outPortalEmbed }
+        );
+
+        if (!sourceMessage.IsSuccess)
+            return Result.FromError(sourceMessage.Error);
+
+        var inPortalEmbed = new Embed(
+            Title: "🌌 Return Portal Created! 🌌",
```

Comment:
Let's remove the emojis here, and make sure the text matches the original implementation.